demo_modules/chaos/chaos.js

```diff
@@ -0,0 +1,138 @@
+// http://jsfiddle.net/TDzVE/ - packed circles
+// http://jsfiddle.net/MattCruikshank/Aq3z8/16/ - chaos theory
+
+/* Copyright 2015 Google Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+const register = require('register');
+const ModuleInterface = require('lib/module_interface');
+const geometry = require('lib/geometry');
+const Rectangle = require('lib/rectangle');
+
+const network = require('network');
+const wallGeometry = require('wallGeometry');
+
+var GOOGLE_COLORS = ['#3369e8', '#d50f25', '#eeb211', '#009925'];
+
+class ChaosServer extends ModuleInterface.Server {
+  willBeShownSoon(container, deadline) {
+  }
+
+  tick(time, delta) {
+  }
+}
+
+class ChaosClient extends ModuleInterface.Client {
+  constructor(config) {
+    super();
+  }
+
+  finishFadeOut() {
+    if (this.surface) {
+      this.surface.destroy();
+    }
+  }
+  
+  hexToRgb(hex) {
+    var result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
+    return result ? {
+        r: parseInt(result[1], 16),
+        g: parseInt(result[2], 16),
+        b: parseInt(result[3], 16)
+    } : null;
+  }  
+
+  willBeShownSoon(container, deadline) {
+    const CanvasSurface = require('client/surface/canvas_surface');
+    this.surface = new CanvasSurface(container, wallGeometry);
+    this.canvas = this.surface.context;
+    this.canvas.fillStyle = 'black';
+    this.canvas.fillRect(0, 0, this.surface.virtualRect.w, this.surface.virtualRect.h);
+    this.imageData = this.canvas.getImageData(0, 0, this.surface.virtualRect.w, this.surface.virtualRect.h);
+    this.raw = new Array(this.surface.virtualRect.w * this.surface.virtualRect.h).fill(0);
```

Comment:
Optional: You *might* be happier with TypedArrays https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray